File: images/create-placeholders.py
#!/usr/bin/env python3
"""
Create placeholder images locally since via.placeholder.com is not accessible
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_placeholder(width, height, text, filename):
    """Create a placeholder image with text"""
    # Create image with light gray background
    img = Image.new('RGB', (width, height), color='#CCCCCC')
    draw = ImageDraw.Draw(img)
    
    # Try to use a default font, fallback to basic if not available
    try:
        # Try to use a larger font
        font_size = min(width, height) // 3
        font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
    except:
        try:
            font = ImageFont.load_default()
        except:
            font = None
    
    # Calculate text position (centered)
    if font:
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
    else:
        text_width = len(text) * 6
        text_height = 11
    
    position = ((width - text_width) // 2, (height - text_height) // 2)
    
    # Draw text
    draw.text(position, text, fill='#666666', font=font)
    
    # Save image
    img.save(filename)
    print(f"Created {filename} ({width}x{height})")

# Create placeholders directory if it doesn't exist
os.makedirs('placeholders', exist_ok=True)

# Create all placeholder images
create_placeholder(24, 18, "?", "placeholders/unknown-flag-small.png")
create_placeholder(48, 36, "?", "placeholders/unknown-flag-large.png")
create_placeholder(800, 200, "No Team Car\nAvailable", "placeholders/no-team-car.png")
create_placeholder(200, 100, "No Logo\nAvailable", "placeholders/no-logo-small.png")
create_placeholder(1320, 400, "No Logo Available", "placeholders/no-logo-large.png")
create_placeholder(1320, 400, "Logo Not Available", "placeholders/logo-not-available.png")
create_placeholder(800, 450, "No Circuit Image\nAvailable", "placeholders/no-circuit-image.png")
create_placeholder(1440, 1080, "No Circuit Background\nAvailable", "placeholders/no-circuit-background.png")
create_placeholder(80, 80, "?", "placeholders/driver-headshot-placeholder.png")

print("\n✅ All placeholder images created!")

File: images/create-placeholders.js
#!/usr/bin/env node

/**
 * Create simple placeholder images using canvas (if available) or skip
 * This is a fallback script for creating placeholder images
 */

const fs = require('fs');
const path = require('path');

// Create placeholders directory
const placeholdersDir = path.join(__dirname, 'placeholders');
if (!fs.existsSync(placeholdersDir)) {
  fs.mkdirSync(placeholdersDir, { recursive: true });
}

// Try to use canvas if available, otherwise create empty files as placeholders
let canvas;
try {
  canvas = require('canvas');
} catch (e) {
  console.log('Canvas module not available. Creating empty placeholder files.');
  console.log('You can manually create these images or install canvas: npm install canvas');
  
  // Create empty files as placeholders
  const placeholders = [
    { name: 'unknown-flag-small.png', size: '24x18' },
    { name: 'unknown-flag-large.png', size: '48x36' },
    { name: 'no-team-car.png', size: '800x200' },
    { name: 'no-logo-small.png', size: '200x100' },
    { name: 'no-logo-large.png', size: '1320x400' },
    { name: 'logo-not-available.png', size: '1320x400' },
    { name: 'no-circuit-image.png', size: '800x450' },
    { name: 'no-circuit-background.png', size: '1440x1080' },
    { name: 'driver-headshot-placeholder.png', size: '80x80' },
  ];
  
  placeholders.forEach(p => {
    const filePath = path.join(placeholdersDir, p.name);
    if (!fs.existsSync(filePath)) {
      fs.writeFileSync(filePath, '');
      console.log(`Created placeholder file: ${p.name} (${p.size})`);
    }
  });
  
  console.log('\nNote: These are empty files. You can download placeholder images from:');
  console.log('https://via.placeholder.com/ or create them manually.');
  process.exit(0);
}

// If canvas is available, create actual images
function createPlaceholder(width, height, text, filename) {
  const img = canvas.createCanvas(width, height);
  const ctx = img.getContext('2d');
  
  // Background
  ctx.fillStyle = '#CCCCCC';
  ctx.fillRect(0, 0, width, height);
  
  // Text
  ctx.fillStyle = '#666666';
  ctx.font = `${Math.min(width, height) / 3}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText(text, width / 2, height / 2);
  
  // Save
  const buffer = img.toBuffer('image/png');
  fs.writeFileSync(path.join(placeholdersDir, filename), buffer);
  console.log(`Created ${filename} (${width}x${height})`);
}

createPlaceholder(24, 18, '?', 'unknown-flag-small.png');
createPlaceholder(48, 36, '?', 'unknown-flag-large.png');
createPlaceholder(800, 200, 'No Team Car\nAvailable', 'no-team-car.png');
createPlaceholder(200, 100, 'No Logo\nAvailable', 'no-logo-small.png');
createPlaceholder(1320, 400, 'No Logo Available', 'no-logo-large.png');
createPlaceholder(1320, 400, 'Logo Not Available', 'logo-not-available.png');
createPlaceholder(800, 450, 'No Circuit Image\nAvailable', 'no-circuit-image.png');
createPlaceholder(1440, 1080, 'No Circuit Background\nAvailable', 'no-circuit-background.png');
createPlaceholder(80, 80, '?', 'driver-headshot-placeholder.png');

console.log('\n✅ All placeholder images created!');

